src/scripts/main.js

```diff
@@ -1,3 +1,341 @@
+
+    cellNode.className = 'field-cell';
+
+    if (cellValue !== 0) {
+      cellNode.classList.add(`field-cell--${cellValue}`);
+    }
+
+    cellNode.innerHTML = cellValue || '';
+  }));
+
+  if (!moveAvaliable) {
+    winMessage.classList.add('hidden');
+
+    loseMessage.classList.remove('hidden');
+  }
+}
+
+function isMoveAvaliable(rowIndex, cellIndex, cellValue) {
+  let result = false;
+
+  if (cellValue === 0) {
+    return true;
+  }
+
+  const toTheRight = cellIndex <= 2
+    ? game[rowIndex][cellIndex + 1]
+    : null;
+
+  const toTheBottom = rowIndex <= 2
+    ? game[rowIndex + 1][cellIndex]
+    : null;
+
+  if (toTheRight === cellValue || toTheBottom === cellValue) {
+    result = true;
+  }
+
+  return result;
+}
+
+function handleNewMove(keyType) {
+  const minIndex = 0;
+  const maxIndex = 3;
+
+  let cellWasMoved = false;
+
+  for (let x = minIndex; x <= maxIndex; x++) {
+    if (keyType === 'ArrowUp') {
+      cellWasMoved = handleMoveUp(x, minIndex, maxIndex) || cellWasMoved;
+    }
+
+    if (keyType === 'ArrowDown') {
+      cellWasMoved = handleMoveDown(x, minIndex, maxIndex) || cellWasMoved;
+    }
+
+    if (keyType === 'ArrowLeft') {
+      cellWasMoved = handleMoveLeft(x, minIndex, maxIndex) || cellWasMoved;
+    }
+
+    if (keyType === 'ArrowRight') {
+      cellWasMoved = handleMoveRight(x, minIndex, maxIndex) || cellWasMoved;
+    }
+  }
+
+  if (cellWasMoved) {
+    populateNewCell();
+  }
+
+  renderGame();
+
+  startButton.innerHTML = 'Restart';
+  startButton.classList.replace('start', 'restart');
+}
+
+function handleMoveUp(x, minIndex, maxIndex) {
+  let swiped;
+  let moveIsPossible = false;
+  const mergedCellCoords = {};
+
+  do {
+    swiped = false;
+
+    for (let y = minIndex; y < maxIndex; y++) {
+      const cell = game[y][x];
+      const nextCellInCol = game[y + 1][x];
+      const cellsCanBeMerged
+        = mergedCellCoords[x] !== y && mergedCellCoords[x] !== y + 1;
+
+      if (nextCellInCol !== 0 && cell === 0) {
+        game[y][x] = game[y + 1][x];
+        game[y + 1][x] = 0;
+
+        swiped = true;
+      }
+
+      if (nextCellInCol === cell && cell !== 0 && cellsCanBeMerged) {
+        game[y][x] += game[y + 1][x];
+        game[y + 1][x] = 0;
+
+        increaseScore(cell + nextCellInCol);
+
+        mergedCellCoords[x] = y;
+
+        swiped = true;
+      }
+    }
+
+    moveIsPossible = moveIsPossible || swiped;
+  } while (swiped);
+
+  return moveIsPossible;
+}
+
+function handleMoveDown(x, minIndex, maxIndex) {
+  let swiped;
+  let moveIsPossible = false;
+  const mergedCellCoords = {};
+
+  do {
+    swiped = false;
+
+    for (let y = maxIndex; y > minIndex; y--) {
+      const cell = game[y][x];
+      const nextCellInCol = game[y - 1][x];
+      const cellsCanBeMerged
+        = mergedCellCoords[x] !== y && mergedCellCoords[x] !== y - 1;
+
+      if (nextCellInCol !== 0 && cell === 0) {
+        game[y][x] = game[y - 1][x];
```

Comment:
the same
replace in all cases
```suggestion
        game[y][x] = nextCellInCol;
```